sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ClassAndMethodName.cs

```diff
@@ -254,21 +254,18 @@ private static bool IsMemberNameValid(string identifierName, out string suggesti
                 return prefix + SuggestFixedCaseName(FirstCharToUpper(input), maxUppercase + 1);
             }
 
-            string suggestionToProcess;
-            if (input.Length == 1 &&
+            var suggestionToProcess = ShouldExcludeFirstLetter()
+                ? FirstCharToUpper(input.Substring(1))
+                : FirstCharToUpper(input);
+
+            return SuggestFixedCaseName(suggestionToProcess, maxUppercase);
+
+            bool ShouldExcludeFirstLetter() =>
+                input.Length == 1 &&
                 !allowInitialI &&
                 startsWithI &&
                 IsCharUpper(input, 0) &&
-                !IsCharUpper(input, 1))
-            {
-                suggestionToProcess = FirstCharToUpper(input.Substring(1));
-            }
-            else
-            {
-                suggestionToProcess = FirstCharToUpper(input);
-            }
-
-            return SuggestFixedCaseName(suggestionToProcess, maxUppercase);
+                !IsCharUpper(input, 1);

```

Comment:
Not changed as a result of this refactoring, but checking `!IsCharUpper(input, 1)` seems to be redundant since `input.Length == 1` has alreayd been checked.

Reply:
This class seems really complex for what it is actually doing but I don't think it's a high priority to rewrite it now.